Launch stages for new 1proposal solution (#1690)
dfinity/portal: docs/developer-docs/integrations/sns/launching/launch-summary-1proposal.md

# Stages of an SNS launch

## Overview

A crucial purpose of an SNS launch is to decentralize the control of an SNS and thereby of
the dapp that the SNS governs.

> Thereby, new tokens must be distributed to a large community to ensure

Author: Thanks. Will update in next commit.

proper decentralization of the voting power. There are of course many ways to do so.

The SNS provides one simple way to achieve this: a developer can hand over their dapp
to the NNS and ask it to create an SNS and start a decentralization swap for it.
The decentralization swap collects ICP from participants and distributes the voting
power of the SNS among participants by swaping the ICP for (staked) SNS tokens.

We first explain the decentralization swap in more detail and then all the stages of
an SNS launch.

## Key concepts

### Decentralization swap

The launch of each SNS includes a separate **decentralization swap canister** that 
is owned and run by the IC.

> In more detail, it is controlled by the NNS root canister.

Author: Thanks. Will update in next commit.

  more information).

  What we have at this stage:

#### Table 1: Canisters

<table>
  <tr>
    <th>Canisters</th>
    <th>Subnet</th>
    <th>Controlled by</th>
    <th>State</th>
  </tr>
  <tr>
    <td>dapp canister(s)</td>
    <td>application subnet</td>
    <td>dapp developer principal</td>
    <td>operational</td>
  </tr>
</table>

### 2. Dapp developers submit NNS proposal to create SNS

This proposal presents all the initial parameters for the SNS as defined in
the first Stage to the NNS community. 
It also defines which dapp canister(s) should be handed over to the SNS.

If successful, at the end of stage, we the following has changed:

#### Table 1: Canisters

<table>
  <tr>
    <th>Canisters</th>
    <th>Subnet</th>
    <th>Controlled by</th>
    <th>State</th>
  </tr>
  <tr>
    <td>dapp canister(s)</td>
    <td>application subnet</td>
    <td>dapp developer principal</td>
    <td>operational</td>
  </tr>
</table>

#### Table 2: NNS Proposals
 <table>
  <tr>
    <th>NNS Proposal</th>
    <th>State</th>
  </tr>
  <tr>
    <td class="light-green-text">SNS creation proposal</td>
    <td class="light-green-text">Open</td>
  </tr>
</table>

> ### 3. Dapp developers add NNS root as co-controller of dapp

Author: Someone pointed out that if devs do that then someone else could send an NNS proposal that is not in the devs interest and therefore it is save to do it in this order. 
I guess the devs co-control the dapp until the execution of the proposal and thus if the proposal is not immediately executed they could still un-assign NNS root again?
I see arguments for both sides ..

> ### 3. Dapp developers add NNS root as co-controller of dapp

Author: Will resolve for now (but happy to continue the discussion)

    <td>application subnet</td>
    <td class="light-orange-text">dapp developer principal, NNS root</td>
    <td>operational</td>
  </tr>
</table>

### 4. The NNS proposal is decided
If the NNS proposal from Stage 2 is adopted, 
then the NNS automatically triggers the remaining stages and thus the creation of
the SNS.

If the proposal is rejected, then the dapp canister(s)' control is given back exclusively
to the original dapp developers.

For the proposal's execution to be successful, the following conditions must also be met:
* NNS root has been added as a co-controller of the dapp to be decentralized (Stage 3
was executed succesfully)
* the SNS-W canister has sufficient cycles.

If these conditions are not met, the proposal will fail immediately and the control is 
given back exclusively to the original dapp developers.

If successful, at the end of this stage the proposal is adopted and the conditions are met.
This means that we are in the following situation:

#### Table 2: NNS Proposals
 <table>
  <tr>
    <th>NNS Proposal</th>
    <th>State</th>
  </tr>
  <tr>
    <td>SNS creation proposal</td>
    <td class="light-orange-text">Adopted</td>
  </tr>
</table>

> ### 5. (Automatically) SNS-W deploys SNS canisters

Reviewer: ```suggestion
### 5. (Automated) SNS-W deploys SNS canisters
```

Author: will do in next commit (here and everywhere)

This results in the following situation:

#### Table 1: Canisters
<table>
  <tr>
    <th>Canisters</th>
    <th>Subnet</th>
    <th>Controlled by</th>
    <th>State</th>
  </tr>
  <tr>
    <td>dapp canister(s)</td>
    <td>application subnet</td>
    <td>dapp developer principal, NNS root</td>
    <td>operational</td>
  </tr>
  <tr>
    <td class="light-green-text">SNS swap</td>
    <td class="light-green-text">SNS subnet</td>
    <td class="light-green-text">NNS root</td>
    <td class="light-green-text">not initialized</td>
  </tr>
 <tr>
    <td class="light-green-text">SNS governance</td>
    <td class="light-green-text">SNS subnet</td>
    <td class="light-green-text">SNS root</td>
    <td class="light-green-text">not initialized</td>
  </tr>
 <tr>
    <td class="light-green-text">SNS ledger</td>
    <td class="light-green-text">SNS subnet</td>
    <td class="light-green-text">SNS root</td>
    <td class="light-green-text">not initialized</td>
  </tr>
 <tr>
    <td class="light-green-text">SNS root</td>
    <td class="light-green-text">SNS subnet</td>
    <td class="light-green-text">SNS governance</td>
    <td class="light-green-text">not initialized</td>
  </tr>
</table>

> ### 6. (Automatically) SNS-W sets SNS root as single controller of dapp
> Once the SNS canisters are deployed, [SNS-W](../introduction/sns-architecture.md#SNS-W)
> sets the SNS root as the single controller of the dapp canister(s).

Reviewer: ```suggestion
### 6. (Automatically) SNS-W sets SNS root as sole controller of dapp
Once the SNS canisters are deployed, [SNS-W](../introduction/sns-architecture.md#SNS-W)
sets the SNS root as the sole controller of the dapp canister(s).
```

Author: done in next commit

For technical reasons, the NNS root canister was added as the co-controller of the dapp
in Stage 3.
Therefore, the SNS-W orchestrates the necessary updates involving NNS root for 
making the appropriate changes.

In detail, this includes two steps:

> * First, SNS-W adds the newly created SNS root cansiter as the dapp canister(s)
>   controller and removes the original dapp developers
>   This is done via a call to NNS root who is the co-controller of the dapp cansiter(s)
>   and thus has the necessary permissions.

Reviewer: If you want to be 100% accurate, I'd suggest writing this

```suggestion
* First, SNS-W removes the original dapp developers from controlling the dapp(s). Next SNS-W adds the newly created SNS root canister as the dapp canister(s) controller. This is done via a call to NNS root who is the co-controller of the dapp canister(s) and thus has the necessary permissions.
```

Author: yes. let's be precise.
will add in next commit.
